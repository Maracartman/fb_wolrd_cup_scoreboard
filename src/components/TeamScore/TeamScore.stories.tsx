import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { TeamScore, TeamScoreProps } from './TeamScore';

const teamScoreProps: TeamScoreProps = {
  home: true,
  team: {
    country: "Venezuela",
    flagHref:
      "https://upload.wikimedia.org/wikipedia/commons/thumb/0/06/Flag_of_Venezuela.svg/1200px-Flag_of_Venezuela.svg.png",
  },
  score: 10,
}

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Example/TeamScore',
  component: TeamScore,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
  argTypes: {}
} as ComponentMeta<typeof TeamScore>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof TeamScore> = (args) => <TeamScore {...args} />

export const  Sample = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
Sample.args = teamScoreProps;
