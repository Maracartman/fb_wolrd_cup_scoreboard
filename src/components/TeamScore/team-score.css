.team-score-container {
    flex-direction: column;
    flex-wrap: wrap;
    justify-content: center;
    text-align: center;
    border: 3px solid rgba(255, 255, 255, 0.733);
    border-radius: 10px;
    padding: 1vw;
    background-color: rgb(4, 39, 7);
}

.ts-data-box {
    flex-direction: row;
    flex-wrap: wrap;
    font-family: monospace;
    font-weight: bold;
    font-size: x-large;
    padding: 3vw;
    color: aliceblue;
    justify-content: center;
    vertical-align: middle;
    text-align: center;
    align-items: center;
}

.ts-status {
    display: flex;
    text-transform: capitalize;
    font-size: 8vw;
    color: rgb(219, 201, 141);
}

.country-name {
    display: flex;
    border-bottom: 3px solid rgba(255, 255, 255, 0.733);
    text-transform: capitalize;
    font-size: 5vw;
}

.ts-flag {
    padding: 1vw;
    display: flex;
    width: 6vw;
    height: 4vw;
}