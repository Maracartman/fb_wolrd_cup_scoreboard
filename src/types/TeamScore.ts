import { Team } from "./Team"

export interface TeamScore {
    team?: Team,
    score?: number
    home?: boolean
}