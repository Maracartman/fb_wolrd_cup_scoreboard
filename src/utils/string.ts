
export const transformToStringScore = (score: number, size?:number): string =>{
    const places = size ?? 2
    return String(score).padStart(places, '0')
}

const StringUtil = {
    transformToStringScore
}

export default StringUtil