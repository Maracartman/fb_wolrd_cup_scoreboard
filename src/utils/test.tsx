import React, { ReactElement } from 'react'
import {RenderOptions, render} from '@testing-library/react'
import TranslationsProvider from 'src/contexts/translations'

const AllTheProviders = ({children}:{children: ReactElement}): ReactElement => (
    <TranslationsProvider>
        {children}
    </TranslationsProvider>
)

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {customRender as render}