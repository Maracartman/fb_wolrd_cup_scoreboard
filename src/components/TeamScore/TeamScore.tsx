import React from 'react';
import { Box, Container} from '@mui/material';
import './team-score.css';
import { Team } from 'src/types/Team';
import { Score } from '../Score/Score';
import { useTranslations } from 'src/contexts/translations';

export interface TeamScoreProps {
  team: Team,
  score: number
  home?: boolean
}

export const TeamScore = ({
  team,
  score,
  home = false
}: TeamScoreProps) => {
  const { homeCopy, visitorCopy } = useTranslations()
  const teamStatus = home ? homeCopy : visitorCopy
  return (
    <Container className='team-score-container'>
      <Box className={'ts-data-box ts-status'}>{teamStatus}</Box>
      <Box className={'ts-data-box country-name'}>
        {team.flagHref && (<img className='ts-flag' alt='Country flag' src={team.flagHref}/>)}
          {team.country}
      </Box>
      <Box className='ts-data-box'><Score score={score}/></Box>
    </Container>
  );
};
