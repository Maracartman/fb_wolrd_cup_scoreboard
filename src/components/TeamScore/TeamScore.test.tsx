import React from "react";
import { screen, render } from "src/utils/test";
import { TeamScore } from "./TeamScore";

const teamScoreProps = {
    home: true,
    team: {
      country: "Venezuela",
      flagHref:
        "https://upload.wikimedia.org/wikipedia/commons/thumb/0/06/Flag_of_Venezuela.svg/1200px-Flag_of_Venezuela.svg.png",
    },
    score: 10,
  }

describe("TeamScore component test suite", () => {
  it("match the snapshot", () => {
    const { container } = render(<TeamScore {...teamScoreProps} />);

    expect(container).toMatchSnapshot();
  });
  it("Display the team's country", () => {
    render(<TeamScore {...teamScoreProps} />)

    expect(screen.getByText("Venezuela")).toBeInTheDocument()
  });

  it("Display the team's score", () => {
    render(<TeamScore {...teamScoreProps} />)

    expect(screen.getByText("10")).toBeInTheDocument()
  });

  it('Display the text "Home" in the team score', () => {
    render(<TeamScore {...teamScoreProps} />)

    expect(screen.getByText("Home")).toBeInTheDocument()
  });

  it('Display the text "Visitor" in the team score', () => {
    render(<TeamScore {...teamScoreProps} home={false}/>)

    expect(screen.getByText("Visit")).toBeInTheDocument()
  });

  it("display the country flag if it's passed", () => {
    render(<TeamScore {...teamScoreProps} />)

    expect(screen.getByAltText("Country flag")).toBeInTheDocument()
  })
});
