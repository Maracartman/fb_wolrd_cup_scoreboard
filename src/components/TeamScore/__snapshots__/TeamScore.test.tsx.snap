// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamScore component test suite match the snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg team-score-container css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="ts-data-box ts-status MuiBox-root css-0"
    >
      Home
    </div>
    <div
      class="ts-data-box country-name MuiBox-root css-0"
    >
      <img
        alt="Country flag"
        class="ts-flag"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/0/06/Flag_of_Venezuela.svg/1200px-Flag_of_Venezuela.svg.png"
      />
      Venezuela
    </div>
    <div
      class="ts-data-box MuiBox-root css-0"
    >
      <div
        class="score MuiBox-root css-0"
      >
        <div
          class="score-text"
        >
          10
        </div>
      </div>
    </div>
  </div>
</div>
`;
